baseapi/apis/rest.py

```diff
@@ -15,13 +15,6 @@
 
 logger = logging.getLogger(__name__)
 DEFAULT_TIMEOUT = 5
```

Comment:
ah right.. having retry requirements be configurable as well. This is fine, but it would mean the eventual consumer will need to supply this retry object and will need to be aware of the backoff_factor
The other fields are fine and self-explanatory:
total: num of retries
status_forcelist: the additional status responses to retry on (additional because connection failures guarantee retries)
allowed_methods: which methods to apply retries on

Reply:
Retries are fully configurable from the consumer when instantiating the client. I've followed the `urllib3/connectionpool.py:HTTPConnectionPool:urlopen()` doco (this is the method that eventually gets called):
* integer number when you simply want to retry connection failures (e.g. first attempt might not connect due to destination server restarting, etc.)
* Retry() object to fully specify which situations to retry on

Reply:
umm.. so I'm guessing `modules/cybersource/cybersource/client.py` would be something along the lines of:
```
...
class CybersourceClient(BaseClient):
    ...
    api_retries = 0 # or Retry(...)
    ...
```
?

Reply:
I'm not following this completely :(
`__init__()` sets up the httpadapter, so having a class var there would not help because the httpadapter has already been mounted?

Reply:
ah wait, the class var is on the client and not RestApi/Api

Reply:
Sorry about the noise here.. I had a slight misunderstanding. I originally intended to have the client set the retries for **all** APIs defined on it.
I've changed this now to simply have each API define its own retries.